### PR TITLE
Remove IsDesktopFacade

### LINKED ISSUE #18697: BuildAllConfigurations and IsDesktopFacade don't play well together
An example of this is System.Security.Principal.Windows.

If you do:
`build.cmd -framework=netfx`
The correct facade ends up under `bin\ref\netfx` from the source build.
The ref gets bin-placed by https://github.com/dotnet/corefx/blob/master/dir.targets#L55

if you do:
`build.cmd -allconfigurations`

The netstandard ref for this assembly ends up under `bin\ref\netfx` which is incorrect. 
The ref gets bin-placed by https://github.com/dotnet/corefx/blob/master/dir.targets#L109

We should consider doing away with IsDesktopFacade completely to help eliminate all this special casing.

### PR DESCRIPTION
**Pull request by ericstj**
Fixes #18697 

/cc @weshaggard 

This removes use of IsDesktopFacade and replaces it with autogenerated reference facades.

I'm running some validation and will update baseline files as necessary in follow up commits.

Please review.

### CODE REVIEW

File: Tools-Override/resolveContract.targets [hunk 1]
```diff
@@ -20,7 +20,7 @@
     </PropertyGroup>
 
     <ItemGroup Condition="'$(HasMatchingContract)' == 'true'">
-      <ResolvedMatchingContract Condition="'$(IsDesktopFacade)' != 'true' and Exists('$(ContractAssemblyPath)')" Include="$(ContractAssemblyPath)" />
```
Comment (weshaggard) [inline Tools-Override/resolveContract.targets]: We can completely eliminate the ProjectReference logic now.

**Comment by weshaggard** [comment]
Looks like the target you want is ResolveMatchingContract, which is what is causing your build failures.

**Comment by ericstj** [comment]
Looks like something is going wrong with type-removals.  The refs point to a type in the same assembly instead of the type in the forwarded assembly.  This is not consistent with what CSC produces.  I'll see if there is a way to rewrite the type-refs as well.

**Comment by ericstj** [comment]
This is still going to fail netfx builds.  @weshaggard I think I'll need your help with rewriting the type-refs when types are removed.  Any idea how to do that in CCI?

**Pull request merged**